### PR TITLE
now that build constaints are supported, add all files

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: gazel/gazel.go [hunk 1]
```diff
@@ -735,16 +735,56 @@ func writeFile(path string, f *bzl.File, exists, dryRun bool) (bool, error) {
...
-		GOARCH:      "amd64",
-		GOOS:        "linux",
-		GOROOT:      build.Default.GOROOT,
-		GOPATH:      build.Default.GOPATH,
-		ReleaseTags: []string{"go1.1", "go1.2", "go1.3", "go1.4", "go1.5", "go1.6", "go1.7"},
-		Compiler:    runtime.Compiler,
-		CgoEnabled:  true,
+func context() *buildCtx {
+	var ctx buildCtx
+	for _, goarch := range []string{"amd64", "arm64"} {
```
Comment (ixdy) [inline gazel/gazel.go]: should we explode the full set of archs? (i.e. add arm, ppc64le, s390x)?

File: gazel/gazel.go [hunk 2]
```diff
@@ -735,16 +735,56 @@ func writeFile(path string, f *bzl.File, exists, dryRun bool) (bool, error) {
...
+		for _, goos := range []string{"linux", "darwin", "windows"} {
+			ctx.ctxs = append(ctx.ctxs, &build.Context{
+				GOARCH:      goarch,
+				GOOS:        goos,
+				GOROOT:      build.Default.GOROOT,
+				GOPATH:      build.Default.GOPATH,
+				ReleaseTags: []string{"go1.1", "go1.2", "go1.3", "go1.4", "go1.5", "go1.6", "go1.7"},
```
Comment (ixdy) [inline gazel/gazel.go]: worth adding go1.8 too?

File: gazel/gazel.go [hunk 3]
```diff
@@ -735,16 +735,56 @@ func writeFile(path string, f *bzl.File, exists, dryRun bool) (bool, error) {
...
+		pkg.Imports = append(pkg.Imports, pkgg.Imports...)
+		pkg.TestGoFiles = append(pkg.TestGoFiles, pkgg.TestGoFiles...)
+		pkg.TestImports = append(pkg.TestImports, pkgg.TestImports...)
+		pkg.XTestGoFiles = append(pkg.XTestGoFiles, pkgg.XTestGoFiles...)
+		pkg.XTestImports = append(pkg.XTestImports, pkgg.XTestImports...)
+	}
+	return &pkg, nil
+}
+
+func (c *buildCtx) ImportDir(dir string, mode build.ImportMode) (*build.Package, error) {
```
Comment (ixdy) [inline gazel/gazel.go:787]: does anything use this?

File: gazel/gazel.go [hunk 4]
```diff
@@ -735,16 +735,56 @@ func writeFile(path string, f *bzl.File, exists, dryRun bool) (bool, error) {
...
+}
+
+type buildCtx struct {
+	ctxs []*build.Context
+}
+
+func (c *buildCtx) Import(path string, dir string, mode build.ImportMode) (*build.Package, error) {
+	var pkg build.Package
+	for _, ctx := range c.ctxs {
+		pkgg, err := ctx.Import(path, dir, mode)
```
Comment (ixdy) [inline gazel/gazel.go:764]: `pkgg` 😕

**Comment by spxtr** [comment]
Is this actually working and ready for review? The commit messages seem to indicate otherwise...

**Comment by spxtr** [comment]
Is this actually working and ready for review? The commit messages seem to indicate otherwise...

**Comment by spxtr** [comment]
Is this actually working and ready for review?

**Pull request closed without merging**